### ISSUE TITLE
MongoDB Connection Count

### ISSUE THREAD

**Issue by cbfhryLee** [RocketChat/Rocket.Chat#6104]
Rocket.Chat Version: 0.52.0
Running Instances: 3 
DB Replicaset OpLog: Active
Node Version: 4.5

Hello. I am a Korean developer who is using Rocketchat well.
I have a question while using Rocketchat and leave a post.

Currently, the following infrastructure is designed.

MongoDB Instance: 3 (Replication)
Rocketchat Instance: 3 (4 core & 1 node)

But the question here is that when no one approaches rocketchat
Here is the log output from MongoDB.

...
...
...
...
2017-02-20T16:30:54.762+0900 I NETWORK  [conn5188] received client metadata from 10.0.0.6:40358 conn5188: { driver: { name: "nodejs", version: "2.2.16" }, os: { type: "Linux", name: "linux", architecture: "x64", version: "3.10.0-514.2.2.el7.x86_64" }, platform: "Node.js v4.5.0, LE, mongodb-core: 2.1.2" }
2017-02-20T16:30:54.763+0900 I NETWORK  [conn5189] received client metadata from 10.0.0.6:40360 conn5189: { driver: { name: "nodejs", version: "2.2.16" }, os: { type: "Linux", name: "linux", architecture: "x64", version: "3.10.0-514.2.2.el7.x86_64" }, platform: "Node.js v4.5.0, LE, mongodb-core: 2.1.2" }
2017-02-20T16:30:56.043+0900 I NETWORK  [conn5190] received client metadata from 10.0.0.4:57430 conn5190: { driver: { name: "nodejs", version: "2.2.16" }, os: { type: "Linux", name: "linux", architecture: "x64", version: "3.10.0-514.2.2.el7.x86_64" }, platform: "Node.js v4.5.0, LE, mongodb-core: 2.1.2" }
2017-02-20T16:30:57.564+0900 I NETWORK  [conn5191] received client metadata from 10.0.0.6:40362 conn5191: { driver: { name: "nodejs", version: "2.2.16" }, os: { type: "Linux", name: "linux", architecture: "x64", version: "3.10.0-514.2.2.el7.x86_64" }, platform: "Node.js v4.5.0, LE, mongodb-core: 2.1.2" }
2017-02-20T16:30:57.980+0900 I NETWORK  [conn5192] received client metadata from 10.0.0.5:38174 conn5192: { driver: { name: "nodejs", version: "2.2.16" }, os: { type: "Linux", name: "linux", architecture: "x64", version: "3.10.0-514.2.2.el7.x86_64" }, platform: "Node.js v4.5.0, LE, mongodb-core: 2.1.2" }
...
...

Logs are continuously appearing at 30 to 40 intervals every 10 seconds.
Is this log normal? Is there really more than 5,000 connections connected to mongoDB?

I have some questions about the above situation.

- MongoDB Primary >> db.serverStatus (). Connections;
=> { "current" : 13, "available" : 51187, "totalCreated" : 5619 }

(((TotalCreated continues to increase.)))

To summarize,

(1) Does the connection log output to MongoDB mean the actual connection count?
(2) If it is an actual Connection Count, can I change the frequency of access attempts within Rocketchat?
(3) If this connection count increases, can connection pool occur in MongoDB?

I look forward to your reply. Thank you for reading the long question.

**Comment by geekgonecrazy** [comment]
@cbfhryLee First off thanks for opening this.  My favorite type of question right now :grin:

So based on your logs this is primarily when a new connection is opened.  When you do:
```
db.serverStatus().Connections
```

Current is the amount of connections currently connected.  In your case 13 between 3 instances is a pretty good number.  This holds with the pattern i've seen with about ~5 per Rocket.Chat instance.

Regarding the high number of created.  When you have users performing actions, the Rocket.Chat instances will open up additional connections from a pool of connections to mongodb.  Once its done with those, it will disconnect the additional connections it opened returning it back to the mongodb pool.

So in short:
1. Shows connections as they are established, but your mongo query is showing current connection count
2. You can tune this by adjusting your mongo connection string. But I wouldn't unless you are having issues.
3. The 51,185 under available are how many connections you have available to connect.  You should be good.

**Comment by cbfhryLee** [comment]
@geekgonecrazy 

Thank you very much for your reply.

In conclusion, I understand that connection pooling does not occur because additional connections are disconnected.

It has helped a lot. Thank you.
